opni-insights-service/app/main.py

```diff
@@ -333,6 +333,59 @@ async def get_logs(start_ts, end_ts):
     return logs_dict
 
 
+async def get_control_plane_components_breakdown(start_ts, end_ts):
+    # Get the breakdown of normal, suspicious and anomolous logs by kubernetes control plane component.
+    kubernetes_components_breakdown_dict = {"Components": []}
+
+    kubernetes_components = [
+        "kubelet",
+        "kube-controller-manager",
+        "kube-api-server",
+        "kube-proxy",
+        "kube-scheduler",
+        "etcd",
+        "k3s-agent",
+        "k3s-server",
+        "rke2-agent",
+        "rke2-server",
+    ]
+    for component_name in kubernetes_components:
```

Comment:
Same feedback here, we should be using elasticsearch aggregations

Reply:
@AmartC I second this. Even though it's a first pass aggregations are easy to do and are **way** more efficient